Agroal Connection Pool

salvadormarcos opened the pull request:
As discussed in #128, follow my implementation with Agroal Connection Pool

flavienlaurent commented on core/src/main/java/com/kumuluz/ee/factories/AgroalDataSourceFactory.java:
```diff
@@ -0,0 +1,171 @@
+        Optional.ofNullable( dscp.getLeakDetectionThreshold() ).map(Duration::ofMillis).ifPresent(pool::leakTimeout);
+
+        // Unknown or non-existent configuration in Agroal
+        Optional.ofNullable( dscp.getMinIdle() ).ifPresent((v) -> logMessageDepreciatedProperty("minIdle") );
```
it can be mapped to minSize no? And is it possible to add all others properties available in AgroalConnectionPoolConfigurationSupplier (initialSize for example)?

salvadormarcos replied:
Yes, this is possible. But for this first moment, I was concerned with supplying only the existing properties in Agroal and HikariCP. And for broken properties, show the message that it is depreciated.

flavienlaurent replied:
I understand but in a production environment it's important to be able to deeply configure the DS ; else performance tuning becomes quite impossible.

gpor0 replied:
Hi @salvadormarcos,

any updates on this, can we mark this implementation as complete?

flavienlaurent replied:
@gpor0 my point of view has not changed, I need to be able to fully configure the DS. If it's merged like this, I will have to fork the repo just to be able to specify few properties on the DS...

gpor0 replied:
Hi @salvadormarcos,
will you improve the implementation with missing properties, as suggested by @flavienlaurent? Do you have any other opened issues?

salvadormarcos replied:
Hi,

I understand @flavienlaurent's need to make detailed settings in DS. However, as commented in [issue 128](https://github.com/kumuluz/kumuluzee/issues/128#issuecomment-465087561), Agroal and Hikari would be kept side by side and Hikari would be used without the JTA component.

I can implement the missing properties, however Hikari does not have many existing properties in Agroal. The only problem would be when Hikari was the DS, several properties would simply be ignored. I believe that this can cause confusion for some users.

@gpor0 I can implement it if you agree with this situation. By the way, it's okay if I give attention to this next week? I am traveling and did not bring my laptop.

gpor0 replied:
Yes @salvadormarcos, please complete the implementation.

salvadormarcos replied:
Done! I analyzed the AgroalConnectionPoolConfigurationSupplier class and the only missing properties were: flushOnClose, initialSize, minSize and idleValidationTimeout.

gpor0 replied:
Great! We will make a review.

salvadormarcos commented:
Hi @gpor0 ,

This option was made available after I implemented the integration, in version 1.5 of Agroal. Yesterday, when I submitted the code, I updated the Kumuluz source, but forgot to update the Agroal dependency. Excuse me :)

I just submitted a new commit to the latest version and the new property.

flavienlaurent commented:
So ? :)

MBJuric commented:
We are testing it extensively. We plan a release this or next week.
